packages/flutter_tools/lib/src/commands/build.dart

```diff
@@ -60,3 +67,107 @@ abstract class BuildSubCommand extends FlutterCommand {
     }
   }
 }
+
+Future<Null> updateVersion({@required BuildInfo buildInfo}) {
+  return updateVersionInPubspec(buildInfo: buildInfo, path: bundle.defaultManifestPath);
+}
+
+@visibleForTesting
+Future<Null> updateVersionInPubspec({@required BuildInfo buildInfo, String path}) async {
+  assert(buildInfo != null, 'buildInfo can\'t be null');
+  final Completer<Null> completer = new Completer<Null>();
+  final Status status = logger.startProgress('Update version in pubspec.yaml...', expectSlowOperation: true);
+
+  // early return, if nothing has to be changed
+  if (buildInfo.buildNumber == null && buildInfo.buildName == null) {
+    completer.complete();
+    status.stop();
+    return completer.future;
+  }
+
+  FlutterManifest flutterManifest;
+  try {
+    flutterManifest = await FlutterManifest.createFromPath(path);
+  } catch (error) {
+    printStatus('Error detected in pubspec.yaml:', emphasis: true);
+    printError('$error');
+    completer.complete();
+    status.stop();
+    return completer.future;
+  }
+
+  if (flutterManifest == null || flutterManifest.isEmpty) {
+    printError('Failed to change the version in pubspec.yaml: Can\'t read pubspec.yaml', emphasis: true);
+    completer.complete();
+    status.stop();
+    return completer.future;
+  }
+
+  if (flutterManifest.buildName == null || flutterManifest.buildNumber == null) {
+    printError('Failed to change the version in pubspec.yaml: `version` in pubspec.yaml not found or it has the wrong format.', emphasis: true);
+    completer.complete();
+    status.stop();
+    return completer.future;
+  }
+
+  final String buildName = buildInfo.buildName ?? flutterManifest.buildName;
+  final int buildNumber = buildInfo.buildNumber ?? flutterManifest.buildNumber;
+
+  final String version = 'version: $buildName+$buildNumber';
+  bool versionLineFound = false;
+
+  final File pubspec = fs.file(path);
+  final String pubspecDirectory = pubspec.dirname;
+
+  final File pubspecTmp = fs.file(fs.path.join(pubspecDirectory, 'pubspec.yaml.tmp'));
+  pubspecTmp.createSync();
+
+  if (pubspec.existsSync() && pubspecTmp.existsSync()) {
```

Comment:
The code herein looks very similar to that in `gradle.dart` -- seems like there's an opportunity to refactor so they share some common code?